hbase-http/src/main/java/org/apache/hadoop/hbase/http/ProxyUserAuthenticationFilter.java

```diff
@@ -150,6 +150,18 @@ static boolean containsUpperCase(final Iterable<String> strings) {
     return false;
   }
 
+  public static String getDoasFromHeader(final  HttpServletRequest request) {
+    String doas = null;
+    final Enumeration<String> headers = request.getHeaderNames();
+    while (headers.hasMoreElements()){
+      String header = headers.nextElement();
+      if (header.toLowerCase().equals("doas")){
+        doas = request.getHeader(header);
```

Comment:
We can break out of the loop here, right ?

Reply:
Yep, that is a good idea, thank you!